user-journey-tests/helpers/testHelpers/biosecurity.js

```diff
@@ -19,31 +19,33 @@ const completeBiosecurityTask = async (radioType) => {
   await taskListPage.selectBiosecurityLink()
   switch (radioType) {
     case 'yes':
-      await keptSeparatelyPage.selectYesAndContinue()
-      await waitForPagePath(grazingPage.pagePath)
-      await grazingPage.selectYesAndContinue()
-      await waitForPagePath(lastGrazedPage.pagePath)
-      await lastGrazedPage.inputLastGrazedAndContinue('2 years')
-      await waitForPagePath(manureAndSlurryPage.pagePath)
-      await manureAndSlurryPage.selectYesAndContinue()
-      await waitForPagePath(howFieldSeparatedPage.pagePath)
+      await keptSeparatelyPage.selectYesAndContinue(grazingPage)
+      await grazingPage.selectYesAndContinue(lastGrazedPage)
+      await lastGrazedPage.inputLastGrazedAndContinue(
+        '2 years',
+        manureAndSlurryPage
+      )
+      await manureAndSlurryPage.selectYesAndContinue(howFieldSeparatedPage)
       await howFieldSeparatedPage.inputSeparatedGrazingAndContinue(
-        'Separate grazing'
+        'Separate grazing',
+        roadsAndTracksPage
+      )
+      await roadsAndTracksPage.selectYesAndContinue(anySharedBuildingsPage)
+      await anySharedBuildingsPage.selectYesAndContinue(
+        minimiseContaminationPage
       )
-      await waitForPagePath(roadsAndTracksPage.pagePath)
-      await roadsAndTracksPage.selectYesAndContinue()
-      await waitForPagePath(anySharedBuildingsPage.pagePath)
-      await anySharedBuildingsPage.selectYesAndContinue()
-      await waitForPagePath(minimiseContaminationPage.pagePath)
       await minimiseContaminationPage.inputMinimiseContaminationAndContinue(
```

Comment:
I guess you could turn this an a lot of other similar functions into a single function inputValueAndContinue if you are passing the value as argument

Reply:
And actually rename to inputValueAndContinueToPage as you are also passing the next page from what I can see?

Reply:
Not sure I'm right here as you might be targeting different elements but worth checking